code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/unit/DeploymentUnit.java

```diff
@@ -359,7 +359,7 @@ protected void createVolume(final Service service, String volumeName, Map<String
         if (template.getExternal()) {
             // external volume should exist, otherwise fail
             volume = context.objectManager.findOne(Volume.class, VOLUME.ACCOUNT_ID, service.getAccountId(),
```

Comment:
This should be findAny(), otherwise madness will ensue if the user manages to get two in there